datasets/penn_treebank/penn_treebank.py

```diff
@@ -0,0 +1,152 @@
+        ),
+    ]
+
+    def _info(self):
+        features = datasets.Features({"tokens": datasets.Value("string")})
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        # TODO: This method is tasked with downloading/extracting the data and defining the splits depending on the configuration
+        # If several configurations are possible (listed in BUILDER_CONFIGS), the configuration selected by the user is in self.config.name
+
+        # dl_manager is a datasets.download.DownloadManager that can be used to download and extract URLs
+        # It can accept any type or nested list/dict and will give back the same structure with the url replaced with path to local files.
+        # By default the archives will be extracted and a path to a cached folder where they are extracted is returned instead of the archive
+        my_urls = {
+            "train": f"{_URL}{_TRAINING_FILE}",
+            "dev": f"{_URL}{_DEV_FILE}",
+            "test": f"{_URL}{_TEST_FILE}",
+        }
+        data_dir = dl_manager.download_and_extract(my_urls)
+        return [
+            datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"filepath": data_dir["train"]}),
+            datasets.SplitGenerator(name=datasets.Split.TEST, gen_kwargs={"filepath": data_dir["test"]}),
+            datasets.SplitGenerator(name=datasets.Split.VALIDATION, gen_kwargs={"filepath": data_dir["dev"]}),
+        ]
+
+    def _generate_examples(self, filepath):
+        """ Yields examples. """
+        # TODO: This method will receive as arguments the `gen_kwargs` defined in the previous `_split_generators` method.
+        # It is in charge of opening the given file and yielding (key, example) tuples from the dataset
+        # The key is not important, it's more here for legacy reason (legacy from tfds)
+        eos_token = "</s>"
+        with open(filepath, encoding="utf-8") as f:
+            for line in f:
+                a = line.split()
+                a.append(eos_token)
+                for id_, word in enumerate(a):
+                    feat = dict()
+                    feat["tokens"] = word
```

Comment:
```suggestion
                    feat["tokens"] = feat.get("tokens", []) + [word]
```
Make sure to add all the words in the line instead of just the last one

Reply:
done